### ISSUE TITLE
Upgrading or installing Python latest version (3.8) does not override pip3 link

### ISSUE THREAD

**Issue by mx-psi** [Homebrew/homebrew-core#57651]
- [X] ran `brew update` and can still reproduce the problem? Yes
- [X] ran `brew doctor`, fixed all issues and can still reproduce the problem? Yes
- [X] ~~ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?~~ no formula failed, see below
- [X] ~~if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?~~ n/a

## What you were trying to do (and why)

*I include links to the CircleCI output when possible, these should be accessible if logged to a CircleCI account.)*

We run a number of CI tests on macOS for the [datadog-agent](https://github.com/DataDog/datadog-agent) repository. For this tests, we use `brew` to install several packages including `python`. 

## What happened (include command output)

Since around 11 am (CET) the tests started failing because the `pip` command could not be found (example [logs available here](https://app.circleci.com/pipelines/github/DataDog/datadog-agent/4096/workflows/c288f2ec-f7b3-4231-9bcc-fd2cf3e851d9/jobs/240489/steps)).

Before the fix, we used [this script](https://github.com/DataDog/datadog-agent-buildimages/blob/ce265d7c0e9e1278eb08b58abd0f179469f4479d/macos/builder_setup.sh#L24-L26) for the setup.
The relevant lines are:

```sh
brew uninstall python@2 || true
brew upgrade python -f || brew install python -f
ln -s /usr/local/bin/pip3 /usr/local/bin/pip
```

We then use `pip` (which links to `pip3`) to install some packages ([see here](https://github.com/DataDog/datadog-agent-buildimages/blob/ce265d7c0e9e1278eb08b58abd0f179469f4479d/macos/build_script.sh#L29)), which is where the CI fails.
After some digging, we believe the likely culprit to be PR #54912. [The logs](https://app.circleci.com/pipelines/github/DataDog/datadog-agent/4100/workflows/b2f47b21-b9ea-4bbd-932c-651fc4e372c4/jobs/240570/parallel-runs/0/steps/0-102) show that some links get overwritten:

<details>
  <summary><code>python@3.8</code> install overwrites</summary>
  <pre>
==> Installing python@3.8
==> Pouring python@3.8-3.8.3_1.high_sierra.bottle.tar.gz
Warning: These files were overwritten during `brew link` step:
/usr/local/bin/2to3
/usr/local/bin/idle3
/usr/local/bin/pydoc3
/usr/local/bin/python3
/usr/local/bin/python3-config
/usr/local/share/man/man1/python3.1
/usr/local/lib/pkgconfig/python3.pc </pre>
</details>

but the `pip3` link does not get overwritten, which seems to be what caused our CI to fail. The fix was to call `pip` as a module, see DataDog/datadog-agent-buildimages#61

Alternatively, we could have fixed it by rewriting the `pip3` link ourselves, since the output from `brew` says

```
You can install Python packages with
 
  /usr/local/opt/python@3.8/bin/pip3 install <package>
```

## What you expected to happen

We expected it to work correctly as it did before 11 am CET ([example of successful log from yesterday](https://app.circleci.com/pipelines/github/DataDog/datadog-agent/4079/workflows/cad9101b-53e7-4378-9b6a-b21bf98781b4/jobs/240368/steps))

## Step-by-step reproduction instructions (by running `brew install` commands)

```
brew upgrade python -f || brew install python -f
pip3
```

**Comment by Bo98** [comment]
Thanks for the detailed report! I have a fix in #57654.

**Comment by Bo98** [comment]
Should be fixed now!
